### PR TITLE
Move defaults and calculations from CLI `*Options` to `*Config` [part 1]

### PR DESCRIPTION
**Pull request by Nashatyrev**
## PR Description

This is a part of effort to introduce Java API: easy node spinning up with `TekuConfiguration`

The aim is to make `TekuConfiguration` (and its members) the primary source of config values and leave CLI `*Options` as a simple facade for command line parsing. The `TekuConfiguration` created either with CLI or with a builder should be the same when the same options are specified. 

I.e. that simple test should work:
```java
public class TekuConfigurationTest extends AbstractBeaconNodeCommandTest {
  @Test
  void test() {
    TekuConfiguration cliConfig = getTekuConfigurationFromArguments();
    TekuConfiguration config = TekuConfiguration.builder().build();
    Assertions.assertThat(config).usingRecursiveComparison().isEqualTo(cliConfig);
  }
}
```

Currently a lot of default option values resides in the `*Options` classes while `*Config` values are either nulls or duplicate default values. The same relates to computable default value s: the most of the logic is handled by the `*Options` classes 

This PR addresses most simple cases and moves defaults (either constant or computed) from `*Options` to `*Config`s:
- Move default config constants from `MetricsOptions` to `MetricsConfig`
- Move default config constants from `BeaconRestApiOptions` to `BeaconRestApiConfig`
- Move default config constants from `InteropOptions` to `InteropConfig`
-  Move default constant and computable values from `P2POptions` and `Eth2NetworkOptions` to `P2PConfig`, `DiscoveryConfig`, `NetworkConfig`, `StorageConfiguration`, `PowchainConfiguration`, `SyncConfig`
- To handle computed default values - add a number of `*Config.Builder` methods `<someOption>Default()` which sets the option value if it's not set explicitly yet

## Fixed Issue(s)

Relates to #4601

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: services/chainstorage/src/main/java/tech/pegasys/teku/services/chainstorage/StorageConfiguration.java [hunk 1]
```diff
@@ -80,12 +85,13 @@ public Spec getSpec() {
   public static final class Builder {
 
     private Eth1Address eth1DepositContract;
-    private StateStorageMode dataStorageMode;
-    private long dataStorageFrequency;
-    private DatabaseVersion dataStorageCreateDbVersion;
+    private StateStorageMode dataStorageMode = StateStorageMode.DEFAULT_MODE;
+    private long dataStorageFrequency = VersionedDatabaseFactory.DEFAULT_STORAGE_FREQUENCY;
+    ;
```
Comment (ajsutton) [inline services/chainstorage/src/main/java/tech/pegasys/teku/services/chainstorage/StorageConfiguration.java]: nit: Stray semicolon.

File: teku/src/main/java/tech/pegasys/teku/config/TekuConfiguration.java [hunk 2]
```diff
@@ -202,6 +203,15 @@ public TekuConfiguration build() {
           eth2NetworkConfigurationBuilder.build();
       final Spec spec = eth2NetworkConfiguration.getSpec();
 
+      Eth1Address depositContractAddress = eth2NetworkConfiguration.getEth1DepositContractAddress();
+      storageConfigurationBuilder.eth1DepositContractDefault(depositContractAddress);
+      powchainConfigBuilder.depositContractDefault(depositContractAddress);
+      powchainConfigBuilder.depositContractDeployBlock(
```
Comment (ajsutton) [inline teku/src/main/java/tech/pegasys/teku/config/TekuConfiguration.java]: Should this be `depositContractDeployBlockDefault`?

**Review by ajsutton — APPROVED** [review]
LGTM. Confirmed that we're still getting default values in the help output which is the main thing.

File: teku/src/main/java/tech/pegasys/teku/config/TekuConfiguration.java (hunk 2 above)
Comment (Nashatyrev) [reply teku/src/main/java/tech/pegasys/teku/config/TekuConfiguration.java]: Right! Good cactch 👍

**Pull request merged**